tests/samples/test_api.py

```diff
@@ -1327,7 +1157,7 @@ async def test_download_reads(
         )
 
     if error != "404_reads":
-        async with AsyncSession(pg) as session:
+        async with AsyncSession(data_layer.samples._pg) as session:
```

Comment:
Just access `pg` directly. It is much clearer than diving through the private attributes of the data layer.